packages/core/stellar/src/zip-scaffold.ts

```diff
@@ -0,0 +1,346 @@
+import JSZip from 'jszip';
+import type { GenericOptions } from './build-generic';
+import type { Contract } from './contract';
+import { printContract, removeCreateLevelAttributes } from './print';
+import { contractsVersionTag } from './utils/version';
+
+const dependencies = {
+  fungible: ['stellar-fungible'],
+  nonFungible: ['stellar-non-fungible'],
+  pausable: ['stellar-pausable', 'stellar-pausable-macros'],
+  upgradable: ['stellar-upgradeable', 'stellar-upgradeable-macros'],
+} as const;
+
+const addDependenciesWith = (dependencyValue: string, dependenciesToAdd: (keyof typeof dependencies)[]) =>
+  dependenciesToAdd.reduce((addedDependency, dependencyName) => {
+    return `${addedDependency}${dependencies[dependencyName].map(cargoDependencies => `${cargoDependencies} = ${dependencyValue}\n`).join('')}`;
+  }, '');
+
+const getDependenciesToAdd = (opts: GenericOptions): (keyof typeof dependencies)[] => {
+  const dependenciesToAdd: (keyof typeof dependencies)[] = [];
+
+  if (opts.kind === 'Fungible') dependenciesToAdd.push('fungible');
+  if (opts.kind === 'NonFungible') dependenciesToAdd.push('nonFungible');
+  if (opts.pausable) dependenciesToAdd.push('pausable');
+  if (opts.upgradeable) dependenciesToAdd.push('upgradable');
+
+  return dependenciesToAdd;
+};
+
+const test = (c: Contract, opts: GenericOptions) => `#![cfg(test)]
+
+extern crate std;
+
+use soroban_sdk::{ ${opts.upgradeable ? 'testutils::Address as _, Address, ' : ''}Env, String };
+
+use crate::contract::{ ${c.name}, ${c.name}Client };
+
+#[test]
+fn initial_state() {
+    let env = Env::default();
+
+    let contract_addr = env.register(${c.name}, (${opts.upgradeable ? 'Address::generate(&env),' : ''}));
+    let client = ${c.name}Client::new(&env, &contract_addr);
+
+    assert_eq!(client.name(), String::from_str(&env, "${c.name}"));
+}
+    
+// Add more tests bellow
+`;
+
+function contractCargo(opts: GenericOptions, scaffoldContractName: string) {
+  return `[package]
+name = "${scaffoldContractName}_contract"
```

Comment:
this should be better kebab-case
```suggestion
name = "${scaffoldContractName.replace(/_/, '-')}-contract"
```